hbase-server/src/main/java/org/apache/hadoop/hbase/coordination/ZKSplitTransactionCoordination.java

```diff
@@ -163,6 +166,10 @@ public void waitForSplitTransaction(final RegionServerServices services, Region
         }
         Thread.sleep(100);
         spins++;
+        if (spins > SPLIT_WAIT_TIMEOUT) {
```

Comment:
I'm trying to wrap my head around the state machine here. So this is more of a question, if we throw here and assuming that the master still transitioned the region racily to "SPLITTING", we will still end up rolling back the transaction right? That seems to be the case based on the logic in "doSplitting()", but just wanted to double check.

Reply:
That is correct. What used to happen so far is, even if in the first try if setData() to ZK fails, we used to propagate BVE up the caller and then rollback splitting, with this patch, we will retry setData() at least 10-15 times before we throw this Exception and hence, the probability of rollback of splitting due to BVE comes down to almost 0

Reply:
Cool, thanks for clarifying.